sdk/dotnet/Pulumi.Automation.Tests/LocalPulumiCmdTests.cs

```diff
@@ -0,0 +1,30 @@
+namespace Pulumi.Automation.Tests
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Xunit;
+
+    using Pulumi.Automation.Commands;
+
+    public class LocalPulumiCmdTests
+    {
+        [Fact]
+        public async Task CheckVersionCommand()
+        {
+            var localCmd = new LocalPulumiCmd();
+            IDictionary<string,string> extraEnv = new Dictionary<string,string>();
+            IEnumerable<string> args = new string[]{ "version" };
+            var result = await localCmd.RunAsync(args, ".", extraEnv
+                // Uncommenting this currently fails the test on v2.23.1:
+                //
+                // Which versions of Pulumi CLI support --event-log param?
+                //
+                // , onEngineEvent: _ev => {}
+            );
```

Comment:
`--event-log` is supported only when using the `PULUMI_DEBUG_COMMANDS` environment variable.

Reply:
Thank you for the pointer! That means I have a bug in env var setting, or something like this. I will investigate further.

Reply:
Although it seems like it is being [passed in](https://github.com/pulumi/pulumi/pull/6640/files#diff-ca2533e085f4365e71a44e2dd166bb28a79ea3610b41aff54b122b1764c5b12eR108-R111) so I'm not sure what the error is here.

Reply:
well he's doing a `version` command in that test and I'd wager that the `--event-log` argument is only supported on the few update commands, right? It's at the very least only relevant on those commands. Unless the other commands are streaming the `StandardOutput` events, I guess

Reply:
Oh yes you're totally right, `--event-log` only works on `up/preview/destroy/refresh`

Reply:
Ah that makes sense! Let me see if we where a test of one of those would belong.

Reply:
maybe in `LocalWorkspaceTests` on the APIs where `UpdateOptions` accepts an `Action<EngineEvent>` property?

Reply:
we could test directly on `LocalPulumiCmd` as well but there is a lot of workspace / project setup that precedes being able to test the engine event listener, setup that `LocalWorkspace` takes care of

Reply:
Let me start grokking that file. That's probably the right place..

Reply:
Right. If this is indirectly covered it might  not separate component tests.

Reply:
There might be value in testing the log file watcher independently, though

Reply:
Leaving "pulumi version" test and added a note which other test covers onEngineEvent, looks like there is decent smoke-test coverage at least already, I was happy to find that.